### ISSUE TITLE
Tab-local current history

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#2051]
Currently there is a global concept of "Current History" (or at least there was as of 16.03ish (we're in between releases)). 
## Issue

If you have multiple Galaxy tabs open, changing history in one, will affect all of them. Change history in one tab, then refresh your history panel in another, and be amazed as it updates to the new history.

When I have multiple analyses going at once, this is an annoyance. Why should I have to switch histories when I could (much more easily) switch tabs. For the antsy users who want to monitor progress on N histories, they can open N different tabs of their preferred browser to watch their histories.
## Solution

I'd like to see current history be tab-local. Changing history in one tab should not affect the work I'm doing in other tabs.

**Comment by martenson** [comment]
It seems toolform page needs to be history_id-aware and toolform could send the request for available datasets with the history_id in it.
What do @guerler and @carlfeberhard think?

**Comment by dannon** [comment]
There is probably a trello card for this somewhere -- this has always been the plan as far as I remember.  (ever since the shift to using the API; this was one of the main points of contention in the whole 'current history' argument, and why we wanted to remove the concept completely)

**Comment by guerler** [comment]
Yes, @martenson that sounds like the right approach. The api endpoint `api/tools/build` would recognize the `history_id` if its passed along while requesting a tool model.
